respx/models.py

```diff
@@ -383,13 +361,13 @@ def match(self, request: Request) -> Optional[Union[Request, ResponseTemplate]]:
         if self.method != request_method.decode():
             return None
 
-        request_url = self.build_url(_request_url)
-        if not self._url:
+        request_url = build_url(_request_url)
+        if not self.url:
             matches = True
-        elif isinstance(self._url, str):
-            matches = self._url == request_url
+        elif isinstance(self.url, str):
+            matches = self.url == request_url
```

Comment:
If `build_url` would return a `httpx.URL`, then our `self.url` is either a that or a `Pattern`.

We could then skip building the request httpx.URL here and do [raw](https://github.com/encode/httpx/blob/master/httpx/_models.py#L273) and compare that with the given raw request url.

Something like...
```py
elif isinstance(self.url, httpx.URL):
    matches = self.url.raw == _request_url
```